projects/netgrif-components-core/src/lib/impersonation/services/impersonation-user-resource.service.ts

```diff
@@ -0,0 +1,27 @@
+import {Injectable} from '@angular/core';
+import {UserResourceService} from '../../resources/engine-endpoint/user-resource.service';
+import {ResourceProvider} from '../../resources/resource-provider.service';
+import {ConfigurationService} from '../../configuration/configuration.service';
+import {Params} from '@angular/router';
+import {Observable} from 'rxjs';
+import {Page} from '../../resources/interface/page';
+import {UserResource} from '../../resources/interface/user-resource';
+import {map} from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ImpersonationUserResourceService extends UserResourceService {
+
+    protected _IMPERSONATION_SERVER_URL: string;
+
+    constructor(protected provider: ResourceProvider, protected _configService: ConfigurationService) {
+        super(provider, _configService);
+        this._IMPERSONATION_SERVER_URL = this.getResourceAddress('impersonation');
+    }
+
+    public search(body: object, params?: Params): Observable<Page<UserResource>> {
+        return this.provider.post$('/impersonate/search', this._IMPERSONATION_SERVER_URL, {query: (body as any).fulltext}, params)
```

Comment:
why is the body defined as an `object` if it needs a `.fulltext` `string` attribute?

a type such as: `{fulltext: string}` would be preferable. It would be even better to define an Interface for this type.

Reply:
Type changed to {fulltext: string} for now.
This method overrides a super class method with the same object type so I just went with it. Should I refactor the super class to use an interface?

Reply:
If you can determine what the correct interface for the method is then this would be the preffered solution. If not, create a Jira issue or at least report it to SD, so that it can be cleaned-up later